executor/shuffle.go

```diff
@@ -412,3 +413,26 @@ func (s *partitionHashSplitter) split(ctx sessionctx.Context, input *chunk.Chunk
 	}
 	return workerIndices, nil
 }
+
+type partitionRangeSplitter struct {
+	byItems    []expression.Expression
+	numWorkers int
+}
+
+func (s *partitionRangeSplitter) split(ctx sessionctx.Context, input *chunk.Chunk, workerIndices []int) ([]int, error) {
+	groupChecker := newVecGroupChecker(ctx, s.byItems)
```

Comment:
Can we avoid creating a group checker each time?

Reply:
yes, good idea